osu.Game/OsuGameBase.cs

```diff
@@ -182,6 +182,13 @@ private void load()
             dependencies.Cache(contextFactory = new DatabaseContextFactory(Storage));
 
             dependencies.Cache(realmFactory = new RealmContextFactory(Storage));
+
+            Host.UpdateThreadPausing += () =>
```

Comment:
What about when changing from single to multi thread? Does the realm context not need to be released on the input thread to be returned to the update thread?

Reply:
Yes, this event should also be run in that case (check framework implementation).